src/js/select.js

```diff
@@ -35,6 +35,24 @@ function initialize(selectEl) {
 
 
 /**
+ * Calculates number of options from a mixed list of options and optgroups.
+ * @param optList - List of option/optgroup.
+ */
+function calculateOptions(optList) {
+  var numOptions = 0;
+
+  Array.from(optList).forEach(function(el) {
```

Comment:
When the order of a loop doesn't matter, you can save some bytes and write a faster loop by iterating backwards:
```javascript
var numOptions = 0,
    i = optList.length,
    el;

while (i--) {
  el = optList[i];
  if (el.tagName === 'OPTGROUP') numOptions += el.children.length;
  else numOptions += 1;
}
```